auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -310,16 +312,51 @@ public TypeSpec createTypeAdapter(ClassName className, ClassName autoValueClassN
     return classBuilder.build();
   }
 
+  private static void buildParameterizedToken(MethodSpec.Builder constructor, TypeName typeArg,
+                                              List<TypeVariableName> typeParams, boolean first) {
+    if (!first) {
+      constructor.addCode(", ");
+    }
+    if (typeArg instanceof ParameterizedTypeName) {
+      ParameterizedTypeName paramTypeArg = (ParameterizedTypeName) typeArg;
+      constructor.addCode("$T.newParameterizedTypeWithOwner(null, $T.class", $Gson$Types.class, paramTypeArg.rawType);
```

Comment:
@swankjesse Do you mind providing some input here?  If someone is using Gson can we consider it safe to use the internal `$Gson$Types` class, since we depend on a specific version of Gson?